step-plans/step-plans-base-artefacts/src/main/java/step/artefacts/handlers/FunctionGroupHandler.java

```diff
@@ -110,7 +125,25 @@ public boolean isOwner(long id) {
 	@Override
 	protected void execute_(ReportNode node, FunctionGroup testArtefact) throws Exception {		
 		Map<String, Interest> additionalSelectionCriteria = tokenSelectorHelper.getTokenSelectionCriteria(testArtefact, getBindings());
-		FunctionGroupContext handle = new FunctionGroupContext(additionalSelectionCriteria);
+		String dockerImage = testArtefact.getDockerImage().get();
+		String containerUser = testArtefact.getContainerUser().get();
+		String containerCommand = testArtefact.getContainerCommand().get();
+
+		Optional<String> dockerImageOptional;
+		if(dockerImage != null && !dockerImage.isEmpty()) {
+			dockerImageOptional = Optional.ofNullable(dockerImage);
+		} else {
+			dockerImageOptional = Optional.empty();
+		}
+
+		Optional<String> containerUserOptional = containerUser != null && !containerUser.isEmpty() ? Optional.ofNullable(containerUser) : Optional.empty();
+		Optional<String> containerCommandOptional = containerCommand != null && !containerCommand.isEmpty() ? Optional.ofNullable(containerCommand) : Optional.empty();
+
+
+		// TODO switch this to a required criteria
+		dockerImageOptional.ifPresent(image -> additionalSelectionCriteria.put("$docker", new Interest(Pattern.compile("true"), false)));

```

Comment:
Should this be changed to a required criteria?

Reply:
@jeromecomte I am not sure, how this works, what would be your suggestion ?

Reply:
like this: new Interest(Pattern.compile("true"), true)